x-pack/plugins/cases/common/api/cases/user_actions/connector.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as rt from 'io-ts';
+import { CaseUserActionConnectorRt, CaseConnectorRt } from '../../connectors';
+import { ActionTypes, UserActionWithAttributes } from './common';
+
+export const ConnectorUserActionPayloadWithoutConnectorIdRt = rt.type({
+  connector: CaseUserActionConnectorRt,
+});
+
+export const ConnectorUserActionPayloadRt = rt.type({
+  connector: CaseConnectorRt,
+});
+
+export const ConnectorUserActionWithoutConnectorIdRt = rt.type({
+  type: rt.literal(ActionTypes.connector),
+  payload: ConnectorUserActionPayloadWithoutConnectorIdRt,
+});
+
+export const ConnectorUserActionRt = rt.type({
+  type: rt.literal(ActionTypes.connector),
+  payload: ConnectorUserActionPayloadRt,
+});
+
+export type ConnectorUserAction = UserActionWithAttributes<rt.TypeOf<typeof ConnectorUserActionRt>>;
```

Comment:
Hey! The problem with `ConnectorUAWithAttributesRt.is` is that it will validate only snake case fields. On the front end, the fields are in camel case and it will fail. That is why I created some type guards that depend on the type & the payload.